app/code/community/Bolt/Boltpay/controllers/OrderController.php

```diff
@@ -196,4 +196,55 @@ public function firecheckoutcreateAction()
             throw $e;
         }
     }
+
+    /**
+     * API to expose order details
+     */
+    public function viewAction()
+    {
+        try {
+            $hmacHeader = $_SERVER['HTTP_X_BOLT_HMAC_SHA256'];
+
+            /* @var Bolt_Boltpay_Helper_Api $boltHelper */
+            $boltHelper = Mage::helper('boltpay/api');
+
+            if (!$boltHelper->verify_hook("{}", $hmacHeader)) {
+                Mage::throwException(Mage::helper('boltpay')->__("Failed HMAC Authentication"));
+            }
+
+            $reference = $this->getRequest()->getParam('reference');
+
+            if (!$reference) {
+                Mage::throwException(Mage::helper('boltpay')->__("Transaction parameter is required"));
+            }
+
+            /** @var Mage_Sales_Model_Order_Payment $orderPayment */
+            $orderPayment = Mage::getModel('sales/order_payment')
+                ->getCollection()
+                ->addFieldToFilter('last_trans_id', $reference)
```

Comment:
@johnmccombs1 
I have two questions here.  

Firstly, what happens if a refund is has been issued on the order?  Does this code still stand up in searching for the `last_trans_id`?

Second, I don't have insight to what the request to this endpoint looks like, but, I am curious as to why you choose to use `transaction id` over `display id/increment_id/merchant order id`?

Wouldn't the latter be the more robust way to lookup an order with perhaps transaction id or order reference/quote_id as a means to validate it?